### PR TITLE
Consolidate runnable variants on DOMManipulationTask into a single Runnable variant.

### PR DESCRIPTION
**Pull request by cbrewster**
Consolidates the runnable variants on `DOMManipulationTask` into a single `Runnable` variant.
Also combines `MainThreadRunnable` into the `Runnable` trait.

I plan on filing a few E-Easy issues after this lands to implement `name` on each of the structs that implement `Runnable`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because refactoring

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12292)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -200,11 +200,9 @@ impl<T: Runnable + Send> Runnable for CancellableRunnable<T> {
 
 pub trait Runnable {
     fn is_cancelled(&self) -> bool { false }
-    fn handler(self: Box<Self>);
-}
-
-pub trait MainThreadRunnable {
-    fn handler(self: Box<Self>, script_thread: &ScriptThread);
+    fn name(&self) -> &'static str { "" }
```
Comment (KiChjang) [inline components/script/script_thread.rs]: I was thinking of "generic runnable" instead of the empty string.

**Comment by KiChjang** [comment]
@bors-servo retry
* #11574

**Comment by KiChjang** [comment]
```
STDERR: Could not terminate process: 21086
  ▶ ERROR [expected OK] /css-transforms-1_dev/html/css-transform-property-existence.htm
  └   → Traceback (most recent call last):
  File "/Users/servo/buildbot/slave/mac-rel-css/build/tests/wpt/harness/wptrunner/executors/base.py", line 149, in run_test
    result = self.do_test(test)
  File "/Users/servo/buildbot/slave/mac-rel-css/build/tests/wpt/harness/wptrunner/executors/executorservo.py", line 133, in do_test
    self.proc.kill()
  File "/Users/servo/buildbot/slave/mac-rel-css/build/python/_virtualenv/lib/python2.7/site-packages/mozprocess/processhandler.py", line 749, in kill
    self.proc.kill(sig=sig)
  File "/Users/servo/buildbot/slave/mac-rel-css/build/python/_virtualenv/lib/python2.7/site-packages/mozprocess/processhandler.py", line 165, in kill
    send_sig(signal.SIGTERM)
  File "/Users/servo/buildbot/slave/mac-rel-css/build/python/_virtualenv/lib/python2.7/site-packages/mozprocess/processhandler.py", line 152, in send_sig
    os.killpg(pid, sig)
OSError: [Errno 1] Operation not permitted
```
Not this again

**Comment by larsbergstrom** [comment]
@bors-servo retry
- #12314

**Pull request merged**